stdlib/2and3/wsgiref/validate.pyi

```diff
@@ -1,50 +1,53 @@
 import sys
-from typing import Any
+from typing import Any, Iterable, Iterator, Optional
+
+from wsgiref.types import WSGIApplication, InputStream, ErrorStream
 
 class WSGIWarning(Warning): ...
 
-def validator(application): ...
+def validator(application: WSGIApplication) -> WSGIApplication: ...
 
 class InputWrapper:
-    input = ...  # type: Any
-    def __init__(self, wsgi_input): ...
-    def read(self, *args): ...
+    input: InputStream
+    def __init__(self, wsgi_input: InputStream) -> None: ...
     if sys.version_info < (3,):
-        def readline(self): ...
+        def read(self, size: int = ...) -> str: ...
+        def readline(self) -> str: ...
     else:
-        def readline(self, *args): ...
-    def readlines(self, *args): ...
-    def __iter__(self): ...
-    def close(self): ...
+        def read(self, size: int) -> bytes: ...
+        def readline(self, size: int = ...) -> bytes: ...
+    def readlines(self, hint: int = ...) -> bytes: ...
+    def __iter__(self) -> Iterable[bytes]: ...
+    def close(self) -> None: ...  # no return
```

Comment:
Why not use `typing.NoReturn`?

Reply:
Oh, I didn't realize that was already in `typing`, since it's not documented. Changing it.